sdk/dotnet/Pulumi/Deployment/Deployment_Invoke.cs

```diff
@@ -81,27 +96,8 @@ private async Task<OutputData<T>> RawInvoke<T>(string token, InvokeArgs args, In
             return data.Value;
         }
 
-        private async Task<SerializationResult> InvokeRawAsync(string token, InvokeArgs args, InvokeOptions? options)
+        private async Task<SerializationResult> InvokeRawAsync(string token, SerializationResult argsSerializationResult, InvokeOptions? options)
         {
-            var label = $"Invoking function: token={token} asynchronously";
-            Log.Debug(label);
-
-            // Be resilient to misbehaving callers.
-            // ReSharper disable once ConstantNullCoalescingCondition
-            args ??= InvokeArgs.Empty;
-
-            // Wait for all values to be available, and then perform the RPC.
-            var argsDict = await args.ToDictionaryAsync().ConfigureAwait(false);
-
-            var keepResources = await this.MonitorSupportsResourceReferences().ConfigureAwait(false);
```

Comment:
Note: slight reordering here, in that MonitorSupportsResourceReferences call will now go before args.ToDictionaryAsync() await, and before   Log.Debug(label); hopefully this is benign.